[TECHNICAL-SUPPORT] LPS-100082

jonathanmccann opened the pull request:
Resending from https://github.com/adolfopa/liferay-portal/pull/1081

I definitely understand not wanting this to be included in SP1. If you have time to review it though we would appreciate it since this is a high priority issue for our client. It does not need to be merged right away but it would be helpful to know if this solution would be accepted or if it needs to be reworked.

Thank you for taking the time to look through this.

From @ChrisKian:

> This is a pretty large issue, so I've tried to break down the commits to be as understandable as possible.  Here's the gist of the issue:
> 
> 1. With the inclusion of LPS-98473, users are now able to subscribe to Journal Articles/Folders/Portlets in a live site.
> 2. With this change came some problems when sending notifications during a publish.
> 3. Since the content update/move/add/delete/etc changes occur in the staging site, there was no way to persist that info and retrieve it when it should actually be sent - during publication.
> 4. Further, certain actions were completely missing (moving folders) and replaced with a notification for each update the article previously had instead.
> 5. My change is to store these live-site notifications, so they can be sent when the time is right
> 6. I did this by utilizing the "delivered" column in the UserNotificationEntry table, which is seemingly unused (always set to true).
> 7. During a publications, I retrieved undelivered UserNotificationEntry objects which belong to the given JournalArticle.  I then updated the delivered column to true, which then would make them appear for the users.
> 8.  For email notifications, this was trickier.  Emails need a lot of context from the SubscriptionSender class, which is populated when the article is modified in the staging site.
> 9. In order to work through this, I was able to pass more values into the UserNotificationEntry's "payload" column, and retrieve them later during publication.

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

jonathanmccann commented:
ci:test:sf

jonathanmccann commented:
ci:test:relevant

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-4-1.liferay.com/job/test-portal-source-format/1677/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-4-1.liferay.com//userContent/jobs/test-portal-source-format/builds/1677/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest(master)/4969/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-2.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/4969/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-5.liferay.com/job/test-portal-source-format/2196/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-5.liferay.com//userContent/jobs/test-portal-source-format/builds/2196/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 2 minutes 41 seconds 845 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/579a116b541592fecec36dcbb8fbff4ba2b96e0d">579a116b541592fecec36dcbb8fbff4ba2b96e0d</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/jonathanmccann/liferay-portal/tree/pull-request-1829-LPS-100082">pull-request-1829-LPS-100082</a><br/>Branch GIT ID: <a href="https://github.com/jonathanmccann/liferay-portal/commit/e18edb22eb108ccea21a0c5ac7b70ae747311920">e18edb22eb108ccea21a0c5ac7b70ae747311920</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-5.liferay.com/job/test-portal-source-format/2196/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-5.liferay.com/userContent/jobs/test-portal-source-format/builds/2196/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 3 minutes 12 seconds 745 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/579a116b541592fecec36dcbb8fbff4ba2b96e0d">579a116b541592fecec36dcbb8fbff4ba2b96e0d</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/jonathanmccann/liferay-portal/tree/pull-request-1829-LPS-100082">pull-request-1829-LPS-100082</a><br/>Branch GIT ID: <a href="https://github.com/jonathanmccann/liferay-portal/commit/e18edb22eb108ccea21a0c5ac7b70ae747311920">e18edb22eb108ccea21a0c5ac7b70ae747311920</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-4-1.liferay.com/job/test-portal-source-format/1677/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-4-1.liferay.com/userContent/jobs/test-portal-source-format/builds/1677/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:relevant - 94 out of 100 jobs passed in 1 hour 29 minutes 30 seconds 561 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/579a116b541592fecec36dcbb8fbff4ba2b96e0d">579a116b541592fecec36dcbb8fbff4ba2b96e0d</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal-ee/tree/master-private">master-private</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal-ee/commit/5746d6699382f799b9ce48ed87e538963df03ff5">5746d6699382f799b9ce48ed87e538963df03ff5</a></p>94 out of 100 jobs PASSED<div><h4>6 Failed Jobs:</h4><ul><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest(master)/4969/"><strike><strong>test-portal-acceptance-pullrequest(master)</strong></strike></a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189931/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/27</strong></strike></a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/266401/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/33</strong></strike></a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/333540/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/35</strong></strike></a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/218304/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</strong></strike></a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/76071/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</strong></strike></a></li></ul></div><details><summary><strong>94 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189937/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316911/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/315410/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/186366/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189935/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/333538/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/10</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316426/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/11</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/320262/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/12</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/266398/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/13</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316425/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/14</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/320261/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/15</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/333536/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/16</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/266399/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/17</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/315408/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/18</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/76072/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/19</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/338319/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/76069/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/20</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/186364/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/21</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/333535/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/22</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316423/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/23</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/76068/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/24</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/186363/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/25</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316422/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/26</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189933/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/28</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189932/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/29</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/333537/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/3</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316428/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/30</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/186369/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/31</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/284330/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/32</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/338322/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/34</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/76074/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/36</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/186368/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/37</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/333539/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/38</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189938/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/39</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/315407/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/4</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/315409/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/40</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/338321/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/41</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/320263/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/42</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/284329/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/43</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/266400/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/44</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316427/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/45</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/76073/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/46</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/338320/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/47</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/284328/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/48</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/186367/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/49</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/76070/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/5</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316430/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/50</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/76076/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/51</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/338324/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/52</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/219141/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/53</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/264922/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/54</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/296704/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/55</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316910/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/56</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/333541/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/57</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/266402/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/58</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/320265/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/59</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189934/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/6</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/315411/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/60</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/76075/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/61</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/328318/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/62</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/296702/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/63</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316909/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/64</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/219140/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/65</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316429/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/66</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/296703/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/67</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/320264/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/68</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/328317/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/69</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316424/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/7</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/264923/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/70</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/328320/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/71</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/312299/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/72</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/186372/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/73</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/338325/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/74</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/219142/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/75</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/296705/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/76</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/316431/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/77</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/76077/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/78</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/315412/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/79</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/186365/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/8</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/266403/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/80</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/338318/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/9</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/186370/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/284331/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/264921/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/320266/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/338323/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/328319/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/284333/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/284332/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/186371/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189940/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189936/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/333542/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189939/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8</a></li></ul></details><h5>For more details click <a href="https://test-1-2.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/4969/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/266401/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/33</a></h5><div><h6>Job Results:</h6><p>6 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/266401//consoleText">AXIS_VARIABLE=0,label_exp=!master #266401</a><ol><li><div><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/266401//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_ContentPagesWithFragmentConfiguration_AddTextConfigurationFragmentToContentPage_">LocalFile.ContentPagesWithFragmentConfiguration#AddTextConfigurationFragmentToContentPage</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-2/1567609275246/test-portal-acceptance-pullrequest(master)/4969/functional-tomcat90-mysql57-jdk8/33/0/LocalFile.ContentPagesWithFragmentConfiguration_AddTextConfigurationFragmentToContentPage/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-2/1567609275246/test-portal-acceptance-pullrequest(master)/4969/functional-tomcat90-mysql57-jdk8/33/0/LocalFile.ContentPagesWithFragmentConfiguration_AddTextConfigurationFragmentToContentPage/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-2/1567609275246/test-portal-acceptance-pullrequest(master)/4969/functional-tomcat90-mysql57-jdk8/33/0/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li></ol></div></li><li><div><h5><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/333540/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/35</a></h5><div><h6>Job Results:</h6><p>23 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/333540//consoleText">AXIS_VARIABLE=1,label_exp=!master #333540</a><ol><li><div><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/333540//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_CPRolesCPWebcontent_AddFolderSiteRoleDefaultPermission_">LocalFile.CPRolesCPWebcontent#AddFolderSiteRoleDefaultPermission</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-2/1567609275246/test-portal-acceptance-pullrequest(master)/4969/functional-tomcat90-mysql57-jdk8/35/1/LocalFile.CPRolesCPWebcontent_AddFolderSiteRoleDefaultPermission/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-2/1567609275246/test-portal-acceptance-pullrequest(master)/4969/functional-tomcat90-mysql57-jdk8/35/1/LocalFile.CPRolesCPWebcontent_AddFolderSiteRoleDefaultPermission/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-2/1567609275246/test-portal-acceptance-pullrequest(master)/4969/functional-tomcat90-mysql57-jdk8/35/1/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li></ol></div></li><li><div><h5><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/76071/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></h5><div><h6>Job Results:</h6><p>0 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/76071//consoleText">AXIS_VARIABLE=0,label_exp=!master #76071</a><pre><code>     [exec] Building JournalArticleLocalization
     [exec] Building JournalArticleResource
     [exec] Building JournalContentSearch
     [exec] Building JournalFeed
     [exec] Building JournalFolder
     [exec] 
     [exec] BUILD SUCCESSFUL in 31s
     [exec] 1 actionable task: 1 executed
     [exec] Gradle build finished at 2019-09-04 15:53:06.265.
     [exec] 
     [exec] See the profiling report at: file:///opt/dev/projects/github/liferay-portal/build/reports/profile/profile-2019-09-04-15-52-35.html
     [exec] A fine-grained performance profile is available: use the --scan option.
     [exec] 
   [delete] Deleting: /opt/dev/projects/github/liferay-portal/build.services.output.txt
   [delete] Deleting: /opt/dev/projects/github/liferay-portal/build.services.diff.txt
     [echo] diff --git a/modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/base/JournalArticleLocalServiceBaseImpl.java b/modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/base/JournalArticleLocalServiceBaseImpl.java
     [echo] index b79cfbd..c66676c 100644
     [echo] --- a/modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/base/JournalArticleLocalServiceBaseImpl.java
     [echo] +++ b/modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/base/JournalArticleLocalServiceBaseImpl.java
     [echo] @@ -641,9 +641,6 @@ public abstract class JournalArticleLocalServiceBaseImpl
     [echo]  		counterLocalService;
     [echo]  
     [echo]  	@Reference
     [echo] -	protected com.liferay.exportimport.kernel.staging.Staging staging;
     [echo] -
     [echo] -	@Reference
     [echo]  	protected com.liferay.portal.kernel.service.ClassNameLocalService
     [echo]  		classNameLocalService;
     [echo]  
[stopwatch] [run.batch.test.action: 4:15.778 sec]
     [echo] .
     [echo] 
     [echo] Detected build services changes. See above build services log for more
     [echo] information. Make sure to commit in all build services results.
      [get] Getting: http://test-1-18/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/76071//consoleText
      [get] To: /opt/dev/projects/github/liferay-portal/20190904155306984.txt
   [delete] Deleting: /opt/dev/projects/github/liferay-portal/20190904155306984.txt
   [delete] Deleting: /opt/dev/projects/github/liferay-portal/null86395588.properties</code></pre></div></li></ol></div></li></ol><hr/><details><summary><strong>Failures in common with <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream(master)">acceptance upstream results</a> at 579a116b541592fecec36dcbb8fbff4ba2b96e0d:</strong></summary><ol><li><div><h5><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189931/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/27</a></h5><div><h6>Job Results:</h6><p>1 Test Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/189931//consoleText">AXIS_VARIABLE=0,label_exp=!master #189931</a><ol><li><div><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/189931//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_PortalSmoke_SmokeWithDeveloperProperties_">LocalFile.PortalSmoke#SmokeWithDeveloperProperties</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-2/1567609275246/test-portal-acceptance-pullrequest(master)/4969/functional-tomcat90-mysql57-jdk8/27/0/LocalFile.PortalSmoke_SmokeWithDeveloperProperties/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-2/1567609275246/test-portal-acceptance-pullrequest(master)/4969/functional-tomcat90-mysql57-jdk8/27/0/LocalFile.PortalSmoke_SmokeWithDeveloperProperties/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-2/1567609275246/test-portal-acceptance-pullrequest(master)/4969/functional-tomcat90-mysql57-jdk8/27/0/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li></ol></div></li><li><div><h5><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/218304/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</a></h5><div><h6>Job Results:</h6><p>3837 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/218304//consoleText">AXIS_VARIABLE=0,label_exp=!master #218304</a><ol><li><div><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/218304//testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest/testScanXMLLog">PortalLogAssertorTest.testScanXMLLog</a><pre><code>junit.framework.AssertionFailedError: 
Unable to extract text from tRThRzmQ
org.apache.tika.exception.TikaException: TIKA-198: Illegal IOException from org.apache.tika.parser.executable.ExecutableParser@639d51f3
	at org.apache.tika.parser.CompositeParser.parse(CompositeParser.java:286)
	at org.apache.tika.parser.CompositeParser.parse(CompositeParser.java:280)
	at org.apache.tika.parser.AutoDetectParser.parse(AutoDetectParser.java:143)
	at org.apache.tika.Tika.parseToString(Tika.java:527)
	at com.liferay.portal.util.FileImpl._parseToString(FileImpl.java:1131)
	at com.liferay.portal.util.FileImpl.extractText(FileImpl.java:447)
	at com.liferay.portal.kernel.util.FileUtil.extractText(FileUtil.java:163)
	at com.liferay.portal.kernel.search.DocumentImpl.addFile(DocumentImpl.java:201)
	at com.liferay.document.library.internal.search.spi.model.index.contributor.DLFileEntryModelDocumentContributor.contribute(DLFileEntryModelDocumentContributor.java:113)
	at com.liferay.document.library.internal.search.spi.model.index.contributor.DLFileEntryModelDocumentContributor.contribute(DLFileEntryModelDocumentContributor.java:60)
	at com.liferay.portal.search.internal.indexer.IndexerDocumentBuilderImpl.lambda$getDocument$1(IndexerDocumentBuilderImpl.java:54)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at com.liferay.portal.search.internal.indexer.IndexerDocumentBuilderImpl.getDocument(IndexerDocumentBuilderImpl.java:52)
	at com.liferay.portal.search.internal.indexer.IndexerWriterImpl.re...</code></pre></div></li></ol></div></li></ol></div></li></ol></details></details></html>

jonathanmccann commented:
ci:test:relevant

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest(master)/4561/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-23.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/4561/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:relevant - 0 out of 1 jobs passed in 21 minutes 47 seconds 218 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/9f435c128bb1d3ca222989fc1219aeb0f93ef20d">9f435c128bb1d3ca222989fc1219aeb0f93ef20d</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal-ee/tree/master-private">master-private</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal-ee/commit/0660bcc11a8369eca9f1d7e6eb8ce16965db9131">0660bcc11a8369eca9f1d7e6eb8ce16965db9131</a></p>0 out of 1 jobs PASSED<div><h4>1 Failed Jobs:</h4><ul><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest(master)/4561/"><strike><strong>test-portal-acceptance-pullrequest(master)</strong></strike></a></li></ul></div><h5>For more details click <a href="https://test-1-23.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/4561/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest(master)/4561/">test-portal-acceptance-pullrequest(master)</a></h5><div><h6>Job Results:</h6><p>0 Jobs Passed.<br/>1 Job Failed.</p></div><pre><code>     [exec] * What went wrong:
     [exec] Could not resolve all files for configuration ':private:apps:portal-mobile-device-detection-fiftyonedegrees-enterprise:portal-mobile-device-detection-fiftyonedegrees-enterprise:dat'.
     [exec] &gt; Could not download com.51degrees.dat.enterprise.gz (com.liferay:com.51degrees.dat.enterprise:3.2-20190801)
     [exec]    &gt; Could not get resource 'http://repository-cdn.liferay.com/nexus/service/local/repo_groups/private/content/com/liferay/com.51degrees.dat.enterprise/3.2-20190801/com.51degrees.dat.enterprise-3.2-20190801.gz'.
     [exec]       &gt; Premature end of Content-Length delimited message body (expected: 135409872; received: 98803712
     [exec] 
     [exec] * Try:
     [exec] Run with --info or --debug option to get more log output. Run with --scan to get full insights.
     [exec] 
     [exec] * Exception is:
     [exec] org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration\$ArtifactResolveException: Could not resolve all files for configuration ':private:apps:portal-mobile-device-detection-fiftyonedegrees-enterprise:portal-mobile-device-detection-fiftyonedegrees-enterprise:dat'.
     [exec] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.rethrowFailure(DefaultConfiguration.java:1054)
     [exec] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.access\$1700(DefaultConfiguration.java:123)
     [exec] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration\$ConfigurationFileCollection.getFiles(DefaultConfiguration.java:1028)
     [exec] 	at org.gradle.api.internal.file.AbstractFileCollection.iterator(AbstractFileCollection.java:76)
     [exec] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.iterator(DefaultConfiguration.java:439)
     [exec] 	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext\$FileTreeConverter.convertInto(DefaultFileCollectionResolveContext.java:217)
     [exec] 	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext.doResolve(DefaultFileCollectionResolveContext.java:117)
     [exec] 	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext.resolveAsFileTrees(DefaultFileCollectionResolveContext.java:88)
     [exec] 	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext\$FileTreeConverter.convertInto(DefaultFileCollectionResolveContext.java:203)</code></pre></div></li></ol><h4>For upstream results, click <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream(master)">here</a>.</h4></details></html>

adolfopa commented:
@jonathanmccann @ChrisKian FYI Brian asked us not to send him anything not directly related to the SP1 until the end of Sept. I'll keep this PR on hold until then.

I have some comments about it, but I need to review it more in depth before asking more changes.  I'll ping you in a couple of days.

Sorry!

jonathanmccann commented:
@adolfopa Thanks for taking the time to review it a bit already and giving a rough timeline. I'll wait for your comments but let me know if you need anything in the mean time.

adolfopa commented on modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/impl/JournalArticleLocalServiceImpl.java:
```diff
@@ -8405,6 +8405,24 @@ else if (action.equals("update")) {
+		catch (Exception e) {
+		}
+
+		if (liveGroupId > 0) {
```
I don't like adding staging specific things to the subscriptions API. Also, internally, both staging and regular subscriptions seem to be handled in the same way. I think we don't need to make a distinction, calling addPersistedSubscribers() should work as well.

adolfopa commented on modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/impl/JournalArticleLocalServiceImpl.java:
```diff
@@ -8413,6 +8431,23 @@ else if (action.equals("update")) {
+					folder.getUuid(), liveGroupId);
+
+				if (folder != null) {
+					subscriptionSender.addPersistedStagingSubscribers(
```
See my previous comment.

adolfopa commented on portal-kernel/src/com/liferay/portal/kernel/util/SubscriptionSender.java:
```diff
@@ -154,6 +161,30 @@ public void flushNotifications() throws Exception {
 
 			_persistestedSubscribersOVPs.clear();
 
+			for (ObjectValuePair<String, Long> ovp :
```
This logic seems to be identical to the non-staging case (other than the boolean). I think it is unnecessary to make the distinction.

Instead of adding a new addPersistedStagingSubscribers() maybe we should rename it to addPersistedDelayedSubscribers() (or something like that, maybe adding an additional boolean value to the regular addPersistedSubscribers() is better) to avoid tying this to staging.

adolfopa commented on portal-kernel/src/com/liferay/portal/kernel/util/SubscriptionSender.java:
```diff
@@ -859,21 +942,41 @@ protected void sendUserNotification(User user) throws Exception {
-				user.getUserId(), portletId,
-				UserNotificationDeliveryConstants.TYPE_PUSH,
-				notificationEventJSONObject);
+			if (notifyImmediately) {
```
Instead of the conditional, why not pass the "notifiyImmediately" value to the method?

jonathanmccann commented:
@adolfopa I believe I've addressed all of your concerns with the newly pushed commits starting with https://github.com/adolfopa/liferay-portal/pull/1084/commits/76242b5d41dade0fb020da22ed975fef2c2792cd.

The only one I was not sure about was keeping the old finder methods. They're not being removed but rather renamed to better reflect their parameters and to avoid duplication. If you want to keep the original names the same what would you suggest we name the new methods? For example the previous finder was named `U_D_A` for `userId`, `delivered`, and `actionRequired`. The new finder being added is for `userId`, `delivered`, and `archived`. In this situation I'm not sure what our normal naming process would be.

adolfopa commented:
Thanks a lot for the changes @jonathanmccann 👍 

> The only one I was not sure about was keeping the old finder methods. They're not being removed but rather renamed to better reflect their parameters and to avoid duplication. If you want to keep the original names the same what would you suggest we name the new methods? For example the previous finder was named `U_D_A` for `userId`, `delivered`, and `actionRequired`. The new finder being added is for `userId`, `delivered`, and `archived`. In this situation I'm not sure what our normal naming process would be.

I'm ok with the new names, but I think we should keep the ones with the old name as well (i.e. add the new ones, but don't remove the old ones from service.xml).  The reason is that some customer may be using those methods in their own custom stuff. It would be fine to remove those methods in a major release, but if we do that in a SP we risk breaking client code, and SPs are supposed to be "safe" upgrades (in theory, anyway).

If you don't mind, I'll make the changes myself and I will forward the pull.

Thanks!

adolfopa commented:
Just started reviewing :)

adolfopa commented:
Pull request submitted to https://github.com/ealonso/liferay-portal/pull/2554. [See changes here.](https://github.com/adolfopa/liferay-portal/compare/6ca8c24b3b00~4...6ca8c24b3b00)
